lib/components/providers/theme/themeProvider.tsx

```diff
@@ -0,0 +1,31 @@
+import { type ReactNode, useEffect, useState } from "react";
+
+import { ThemeContext, initialThemeState } from "./themeContext";
+
+type Props = {
+    children: ReactNode;
+};
+
+export const ThemeProvider = ({ children }: Props) => {
+    const [theme, setTheme] = useState(initialThemeState.theme);
+
+    const localStorage = globalThis.window?.localStorage;
```

Comment:
```suggestion    
    const [theme, setTheme] = useState(savedThemeLocal  ?? initialThemeState.theme);
```

Reply:
Not sure if this works but the idea behind it is that if the `globalTheme` from localStorage has value then it will be initalState if not then `initialThemeState.theme` is the initialState.